### PR TITLE
refactor: 모달 계열 back drop 구조 및 애니메이션 개선

### PR DESCRIPTION
**Pull request by chanwukim**
## 📝 개요

스크린샷 2023-09-24 오후 4 09 50

- 및 모달 계열 컴포넌트들 애니메이션들을 좀더 앱스럽게 개선했습니다

### modal

https://github.com/oduck-team/oduck-client/assets/105474635/47a23243-6b1f-4b94-b81d-69c4979c98e7

https://github.com/oduck-team/oduck-client/assets/105474635/5f4e057a-eb66-43e3-9565-6767a491ba60

### drawer (디자인 figma 반영)

https://github.com/oduck-team/oduck-client/assets/105474635/364fe58b-b720-40a3-aa5f-a8cd85d80881

https://github.com/oduck-team/oduck-client/assets/105474635/e611a7d1-1402-4aa5-8521-bf21f09e2508

https://github.com/oduck-team/oduck-client/assets/105474635/3fe36113-d8b9-4111-ad06-58624fb6eb8d

https://github.com/oduck-team/oduck-client/assets/105474635/fae81654-2ccd-4a2a-8b1d-9af7246ead94

### bottom sheet

- Footer css 반영 전 녹화영상입니다!

- 만약 높이가 view port를 초과할경우

https://github.com/oduck-team/oduck-client/assets/105474635/b958f0db-6612-4329-b3f5-3e432fd520cb

- 아닐경우

https://github.com/oduck-team/oduck-client/assets/105474635/3e0f2c5c-3076-4477-8a39-a0fd64b1e784

### 기타
- [feat: useWindowSize 훅](https://github.com/oduck-team/oduck-client/commit/7190927eb0d6b200a3791ea488c65b79d7d6da93)
- `style/z-index.ts` -> `style/zIndex.ts`
- `types/emotion.d.ts` 주석 작성
- `styles/container.ts` 주석 작성
## 🔗 관련 이슈

#140 
#136

## ➕ 기타
모바일영상 용량이 커서 첨부 실패했습니다 🙇🏻‍♂️

### CODE REVIEW

**Comment by chanwukim** [comment]
모바일 bottom sheet

https://github.com/oduck-team/oduck-client/assets/105474635/732c582d-ef81-41b6-8cd1-3408a0b45611

**Comment by presentKey** [comment]
resize 이벤트에 throttle 적용하면 좋을 거 같습니다. resize 및 scroll 이벤트는 작은 변경에도 이벤트 호출이 많이  발생합니다!

File: src/components/Portal/index.tsx [hunk 1]
```diff
@@ -1,15 +1,24 @@
+import { useEffect, useState } from "react";
 import { createPortal } from "react-dom";
 
 import { StrictPropsWithChildren } from "@/types";
 
 interface PortalProps {
-  elementId: string;
+  elementId?: string;
```
Comment (imdaxsz) [inline src/components/Portal/index.tsx]: 이 파일 내에서 더 이상 사용되지 않는 것 같은데 남겨두신 이유가 있을까요?

Comment (chanwukim) [reply src/components/Portal/index.tsx]: 실수네요 지우겠씁니다 👍

**Review by imdaxsz — APPROVED** [review]
고생하셨습니다~~ 많이 배우네요 ㅎㅎ 👍

**Pull request merged**